test/fail_compilation/fail21849.d

```diff
@@ -0,0 +1,13 @@
+// https://issues.dlang.org/show_bug.cgi?id=21849
+// REQUIRED_ARGS: -verrors=context -vcolumns
+/* TEST_OUTPUT:
+---
+fail_compilation/fail21849.d(12,33): Error: cannot implicitly convert expression `cast(ushort)65535u` of type `ushort` to `byte`
```

Comment:
One enhancement request would be to also put in the display column number `(12, 30)`, not the raw byte column here.  But that would mean fixing `Loc.toChars()` to also be aware of multibyte characters.